src/components/number-input/number-input.scss

```
@@ -14,4 +14,8 @@ $css--plex: true !default;
   // the base text-input class has a 100% so we need to propagate it up to the host element
   width: 100%;
   outline: none;
+
+  .bx--form-requirement.bx--hidden {
```

Wondered if we can use hidden attribute (`[hidden]` selector) here.

---

I actually tried that first! For some reason it didn't hide slotted content ... I'm not sure if that's a pattern that's been successful elsewhere though

---

Just a hunch... Any chance it was due to missing `?` prefix at `?hidden="${validity !== NUMBER_INPUT_VALIDATION_STATUS.EXCEEDED_MINIMUM}"`?

```suggestion
  .bx--form-requirement[hidden] {
```

I could make it work with the combination of above two somehow. I may be wrong, though.

---

Hmm ... definitely had the `?` 🤔 I didn't have the styles when I tried it, so that would definitely make it work